File: cpp/vm/monkey/repl.cpp
#include "repl.h"
#include <iostream>

void Repl::loop() {
    while(1) {
        std::string line;
        std::cout << ">> ";
        getline(std::cin, line);
        if (line == "exit()") break;
        parser.load(line);
        parser.buildProgramStatement(program);
    }
}

File: java/VMPrototype1/src/main/java/ro/adrianus/prototype/repl/Repl.java
package ro.adrianus.prototype.repl;

import ro.adrianus.prototype.ast.AstProgram;
import ro.adrianus.prototype.evaluator.Evaluator;
import ro.adrianus.prototype.parser.Parser;

import java.util.Scanner;

public class Repl {
    private Scanner scanner = new Scanner(System.in);
    private Parser parser;
    private AstProgram program;
    private Evaluator evaluator;

    public Repl(Parser parser, AstProgram program, Evaluator evaluator) {
        this.parser = parser;
        this.program = program;
        this.evaluator = evaluator;
    }

    public void executeSource(String source) {

    }

    public void loop() {
        while(true) {
            System.out.print(">> ");
            String line = scanner.nextLine();
            if (line.equals("exit()")) break;
            parser.load(line);
            program.reset();
            parser.buildProgramStatements(program);
            //evaluator.eval(program, env);
        }
    }
}
